robot_smach_states/src/robot_smach_states/human_interaction/find_people_in_room.py

```diff
@@ -0,0 +1,532 @@
+            requirements posed using the properties, query_entity_designator, look distance and strict arguments
+        :param attempts: (int) (default: 1) Max number of search attempts
+        :param search_timeout: (float) (default: 60) maximum time the robot is allowed to search
+        :param look_range: (tuple of size 2) (default: (-np.pi/2, np.pi/2)) from what to what head angle should the
+        robot search
+        :param look_steps: (int) (default: 8) How many steps does it take in that range
+        """
+        smach.State.__init__(self, outcomes=['found', 'failed'])
+
+        self._robot = robot
+
+        self._properties = properties
+        self._look_distance = look_distance
+        self._speak = speak
+        self._strict = strict
+        self._nearest = nearest
+        self._attempts = attempts
+
+        self._search_timeout = search_timeout
+
+        self._look_angles = np.linspace(look_range[0], look_range[1], look_steps)
+
+        if found_people_designator:
+            ds.is_writeable(found_people_designator)
+            ds.check_type(found_people_designator, [Entity])
+        self._found_people_designator = found_people_designator
+
+        if query_entity_designator:
+            ds.check_type(query_entity_designator, Entity)
+        self._query_entity_designator = query_entity_designator
+
+    def execute(self, userdata=None):
+        look_angles = None
+        person_label = None
+
+        if self._properties:
+            try:
+                person_label = self._properties["id"]
+                ds.check_type(person_label, "str")
+                person_label = person_label.resolve() if hasattr(person_label, 'resolve') else person_label
+
+                rospy.loginfo("Trying to find {}".format(person_label))
+            except:
+                # The try except is to check if a named person is queried for
+                # in the properties. If there is none then exception is raised
+                # and nothing is to be done with it
+                pass
+
+        if self._speak:
+            self._robot.speech.speak(
+                "{}please look at me while I am looking for you".format(
+                    person_label+', ' if person_label else ''),
+                block=False)
+
+        start_time = rospy.Time.now()
+
+        head_goals = [kdl_conversions.VectorStamped(x=self._look_distance * math.cos(angle),
+                                                    y=self._look_distance * math.sin(angle),
+                                                    z=1.5,
+                                                    frame_id="/%s/base_link" % self._robot.robot_name)
+                      for angle in self._look_angles]
+
+        i = 0
+        attempts = 0
+
+        rate = rospy.Rate(2)
+        while not rospy.is_shutdown() and attempts < self._attempts and (rospy.Time.now() - start_time).to_sec() < self._search_timeout:
```

Comment:
The way the function is set up now it will return the list of found people as soon as it has found at least one person meeting the requirements. This means that if the robot sees someone on his right on the first head goal, someone standing on the left of the robot might not be included in the returned list. Is this the intended functionality or should the returned list be complete?